packages/cloud-cognitive/src/components/AddSelect/AddSelect.js

```diff
@@ -55,17 +56,32 @@ export let AddSelect = forwardRef(
     };
 
     // hooks
+    const [filteredItems, setFilteredItems] = useState([]);
     const [singleSelection, setSingleSelection] = useState('');
     const [multiSelection, setMultiSelection] = useState([]);
     const [searchTerm, setSearchTerm] = useState('');
 
+    // save items locally
+    useEffect(() => {
+      setFilteredItems(items);
+    }, [items]);
+
```

Comment:
I'm a bit wary of having two useEffects both of which call 'setFilteredItems'. They are both dependent on `items`, and if `items` changes I don't think we can be certain which order the hooks will run in. This could mean that when there is a filter in effect, programmatically changing the set of items causes the filtering to be temporarily "lost" if the second useEffect triggers first (applying filtered items) but then this one triggers second, applying the unfiltered items list.

I think it might be safer to just have the second useEffect, and make sure that if there is no `searchTerm` set then results is just set equal to the unfiltered items. Then that hook is responsible for making sure the list is set, filtered or unfiltered.

Reply:
ahh yes i see what you mean. yes that does sound like a good solution.